Assets/Scripts/Guns/GunController.cs

```diff
@@ -53,25 +54,29 @@ private void HandleFire()
     {
       if (Input.GetButtonDown("Fire1"))
       {
-        activeGun.OnMoveDown(transform, attachedCamera);
+        ActiveGun.OnMoveDown(transform, _attachedCamera);
       }
       else if (Input.GetButtonDown("Fire2"))
       {
-        activeGun.OnRightClick(attachedCamera);
+        ActiveGun.OnRightClick(_attachedCamera);
       }
     }
 
     private void HandleChangeGun()
     {
-      var pressedGunKey = guns
+      var pressedGunKey = _guns
           .Keys
           .FirstOrDefault(Input.GetKeyDown);
 
       if (pressedGunKey == KeyCode.None) 
         return;
 
-      activeGun.OnSwitchedAway();
-      activeGun = guns[pressedGunKey];
+      if (ActiveGun == _guns[pressedGunKey])
+        return;
+
+      ActiveGun.OnSwitchedAway();
+      ActiveGun = _guns[pressedGunKey];
+      _toolGunController.SetGunModeText(ActiveGun.GunName);
```

Comment:
duplicated method call (already calling it in property ActiveGun {set}

Reply:
fixed